mod_reforged/hooks/crafting/crafting_manager.nut

```diff
@@ -0,0 +1,75 @@
+::Reforged.HooksMod.hook("scripts/crafting/crafting_manager", function(q) {
+	//
+	q.getQualifiedBlueprints = @(__original) function()
+	{
+		local settingValue = ::Reforged.Mod.ModSettings.getSetting("CraftingBlueprintVisibility").getValue();
+		switch (settingValue)
+		{
+			case "Always":
+			{
+				local ret = [];
+				foreach (b in this.m.Blueprints)
+				{
+					local oldTimesCrafted = b.m.TimesCrafted;
+					b.m.TimesCrafted = 1;	// We want to show all recipes and by setting TimesCrafted to 1, we trick Vanilla to do just that
+					if (b.isQualified())
+					{
+						ret.push(b);
+					}
+					b.m.TimesCrafted = oldTimesCrafted;
+				}
+				return ret;
+			}
+			case "One Ingredient Available":
+			{
+				local ret = [];
+				foreach (b in this.m.Blueprints)
+				{
+					if (b.isPartlyCraftable())
+					{
+						local oldTimesCrafted = b.m.TimesCrafted;
+						b.m.TimesCrafted = 1;	// We want to force-show this recipe, unless there is another custom visibilty rule
+						if (b.isQualified())
+						{
+							ret.push(b);
+						}
+						b.m.TimesCrafted = oldTimesCrafted;
+					}
+				}
+				return ret;
+			}
+			case "All Ingredients Available":
+			{
+				local ret = [];
+				foreach (b in this.m.Blueprints)
+				{
+					local oldTimesCrafted = b.m.TimesCrafted;
+					b.m.TimesCrafted = 0;	// We do not want to show recipes, just because we crafted them once before, so we trick Vanilla to not show those
+					if (b.isQualified())
+					{
+						ret.push(b);
+					}
+					b.m.TimesCrafted = oldTimesCrafted;
+				}
+				return ret;
+			}
+			case "Vanilla":		// Do nothing, because this is equal to the catch-all return at the very end
+		}
+
+		return __original();
```

Comment:
Move this return statement into the Vanilla case, and move it up top as the first option. This makes it clearer that on the "Vanilla" setting, we just run the original function.
Also add a default: case where you throw with a message that the case was not recognized. This is a failsave against future regressions.